### PR TITLE
[MM-48889] Catch error when attempting to delete draft

### PR DESCRIPTION
**Pull request by mylonsuren**
#### Summary

Catches errors when deleting drafts. Currently we delete a draft when sending a post, so it will return an error if there does not exist a draft in the server to delete. 

Follow-up ticket created here to limit the number of `deleteDraft` calls being made: [MM-48893](https://mattermost.atlassian.net/browse/MM-48893)

#### Ticket Link

[MM-48889](https://mattermost.atlassian.net/browse/MM-48889)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by amyblais** [comment]
@M-ZubairAhmed @ashishbhate Please help prioritize this PR review.

File: actions/views/drafts.ts [hunk 1]
```diff
@@ -48,7 +48,12 @@ export function removeDraft(key: string, channelId: string, rootId = '') {
 
         if (syncedDraftsAreAllowedAndEnabled(state)) {
             const connectionId = getConnectionId(getState() as GlobalState);
-            await Client4.deleteDraft(channelId, rootId, connectionId);
+
+            try {
+                await Client4.deleteDraft(channelId, rootId, connectionId);
+            } catch (error) {
+                return error;
```
[the remark this answers is not in the document]
Comment (ashishbhate) [reply actions/views/drafts.ts:55]: I agree, we should log this. From the linked the follow-up ticket, it appears we often send a delete draft request for drafts that don't actually exist? 
In that case, we should probably also update the server to not return an error when attempting to delete a draft that doesn't exist.

[the remark this answers is not in the document]
Comment (mylonsuren) [reply actions/views/drafts.ts]: Yeah, @ashishbhate I'm going to create a separate server PR to not return errors when when there doesn't exist a draft in the server to delete. I've update the return value for this for now.

**Comment by mylonsuren** [comment]
@ashishbhate This should be good for another review, a PR for handling the 404 error has been created here: https://github.com/mattermost/mattermost-server/pull/21818

**Review by ashishbhate — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thanks @mylonsuren 
Tested, looks good to merge.
Verified draft posting/deleting works as expected - few errors present, that will be resolved by the server PR 21818
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/272866-04fc7bf-onprem-ent-webapp-pr-11738-mm-ee-test:04fc7bf/mochawesome.html) - no PR related failures

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**